src/styles.ts

```diff
@@ -57,29 +99,45 @@ export class StyleCompiler extends EventEmitter {
       }));
   }
 
-  private postprocess(processed: any) {
+  private postprocess(output: string, processed: any) {
     return new Promise((resolve, reject) => {
-      postcss(this.config.postcss.map(([pluginName, pluginConfig]) =>
-        require(pluginName)(pluginConfig)))
-        .process(processed.css, {map: {inline: false}, to: this.config.output})
+      postcss((this.config.postcss || []).map(([pluginName, pluginConfig]) => {
+        const plugin = require(pluginName);
+        if (pluginConfig) {
+          return plugin(pluginConfig);
+        }
+        return plugin;
+      }))
+        .process(processed.css, {
+          map: this.config.sourceMaps && {inline: false},
+          to: output
+        })
         .then(
           (result: any) => resolve(result),
           reject
         );
     });
   }
 
-  private write(processed: any) {
-    return Promise.all([
-      utils.writeFile(this.config.output, processed.css),
-      utils.writeFile(`${this.config.output}.map`, processed.map)
-    ]);
+  private write(output: string, processed: any) {
+    const content = this.config.asESModule ? `export default \`${processed.css}\`;` : processed.css;
+    return Promise.all(
+      [utils.writeFile(output, content)]
+        .concat(this.config.sourceMaps ? [utils.writeFile(`${output}.map`, processed.map)] : []));
+  }
+
+  private compile(entry: string) {
+    const outputFile = this.config.output
+      || `${this.config.outDir}/${entry.replace(/\.(s[ac]ss)$/, this.config.asESModule ? ".$1.js" : ".css")}`;
```

Comment:
Please always use `path.join` instead of plain `/` when dealing with paths - ensures cross-platform compatibility.